<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Reports and functions used in the Genomics Platform Group at the University of Melbourne Centre for Cancer Research.">
<title>Genomics Platform Group Reporting â€¢ gpgr</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Genomics Platform Group Reporting">
<meta property="og:description" content="Reports and functions used in the Genomics Platform Group at the University of Melbourne Centre for Cancer Research.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">gpgr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.2.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/devnotes.html">Developers</a>
    <a class="dropdown-item" href="articles/purple.html">PURPLE</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/umccr/gpgr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><ul>
<li>
<a href="#books-gpgr---genomics-platform-group-reporting">ðŸ“š gpgr - Genomics Platform Group Reporting</a>
<ul>
<li><a href="#installation">Installation</a></li>
<li>
<a href="#main-modules">Main modules</a>
<ul>
<li><a href="#id_-purple">ðŸ”® PURPLE</a></li>
<li><a href="#id_-sash">sash</a></li>
</ul>
</li>
<li><a href="#id_-developers">ðŸ¥³ Developers</a></li>
<li><a href="#id_-cli">ðŸ’» CLI</a></li>
</ul>
</li>
</ul>
<!-- README.md is generated from README.Rmd. Please edit that file --><div class="section level1">
<div class="page-header"><h1 id="books-gpgr---genomics-platform-group-reporting">ðŸ“š gpgr - Genomics Platform Group Reporting<a class="anchor" aria-label="anchor" href="#books-gpgr---genomics-platform-group-reporting"></a>
</h1></div>
<p>Contains reports and functions used in the Genomics Platform Group at the University of Melbourne Centre for Cancer Research.</p>
<ul>
<li>Docs: <a href="https://umccr.github.io/gpgr/" class="uri">https://umccr.github.io/gpgr/</a>
</li>
</ul>
<p><a href="https://anaconda.org/umccr/r-gpgr" class="external-link"><img src="https://anaconda.org/umccr/r-gpgr/badges/version.svg" alt="Conda install"></a> <a href="https://anaconda.org/umccr/r-gpgr" class="external-link"><img src="https://anaconda.org/umccr/r-gpgr/badges/latest_release_date.svg" alt="Conda install"></a></p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"umccr/gpgr"</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>Or if used inside a conda environment:</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">conda</span> install r-gpgr <span class="at">-c</span> umccr <span class="at">-c</span> conda-forge <span class="at">-c</span> bioconda</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="main-modules">Main modules<a class="anchor" aria-label="anchor" href="#main-modules"></a>
</h2>
<div class="section level3">
<h3 id="crystal_ball-purple">ðŸ”® PURPLE<a class="anchor" aria-label="anchor" href="#crystal_ball-purple"></a>
</h3>
<ul>
<li>Read and process output files from the <code>PURPLE</code> purity/copy number estimator tool from the Hartwig Medical Foundation (<a href="https://github.com/hartwigmedical/hmftools/tree/master/purple" class="external-link uri">https://github.com/hartwigmedical/hmftools/tree/master/purple</a>). See vignette at <a href="https://umccr.github.io/gpgr/articles/purple.html" class="uri">https://umccr.github.io/gpgr/articles/purple.html</a>.</li>
</ul>
</div>
<div class="section level3">
<h3 id="sash">sash<a class="anchor" aria-label="anchor" href="#sash"></a>
</h3>
<ul>
<li>Generate a HTML report with results from the <code>sash</code> WGS post-processing workflow
<ul>
<li>
<a href="https://github.com/scwatts/sash" class="external-link uri">https://github.com/scwatts/sash</a>. See the <a href="#cli">CLI</a> section below for options.</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="partying_face-developers">ðŸ¥³ Developers<a class="anchor" aria-label="anchor" href="#partying_face-developers"></a>
</h2>
<p>See <a href="https://umccr.github.io/gpgr/articles/devnotes.html" class="uri">https://umccr.github.io/gpgr/articles/devnotes.html</a> for developer notes.</p>
</div>
<div class="section level2">
<h2 id="computer-cli">ðŸ’» CLI<a class="anchor" aria-label="anchor" href="#computer-cli"></a>
</h2>
<p>A <code>gpgr</code> command line interface is available for convenience.</p>
<ul>
<li>If youâ€™re using the conda package, the <code>gpgr.R</code> command will already be set up inside an activated conda environment.</li>
<li>If youâ€™re <em>not</em> using the conda package, you need to export the <code>gpgr/inst/cli/</code> directory to your <code>PATH</code> in order to use <code>gpgr.R</code>.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="va">gpgr_cli</span><span class="op">=</span><span class="va">$(</span><span class="ex">Rscript</span> <span class="at">-e</span> <span class="st">'x = system.file("cli", package = "gpgr"); cat(x, "\n")'</span> <span class="kw">|</span> <span class="fu">xargs</span><span class="va">)</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${gpgr_cli}</span><span class="st">:</span><span class="va">${PATH}</span><span class="st">"</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>gpgr.R <span class="sc">--</span>version</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>gpgr.R <span class="dv">2</span>.<span class="fl">1.0</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#-----------------------------------#</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>gpgr.R <span class="sc">--</span>help</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>usage<span class="sc">:</span> gpgr.R [<span class="sc">-</span>h] [<span class="sc">-</span>v] {canrep} ...</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>UMCCR Genomics Platform Group Reporting</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>positional arguments<span class="sc">:</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  {canrep}       sub<span class="sc">-</span>command help</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>    canrep       UMCCR Cancer Report.</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>options<span class="sc">:</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  <span class="sc">-</span>h, <span class="sc">--</span>help     show this help message and exit</span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>  <span class="sc">-</span>v, <span class="sc">--</span>version  show program<span class="st">'s version number and exit</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="st">#-----------------------------------#</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="st">#------- Cancer Report -------#</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="st">gpgr.R canrep --help</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="st">usage: gpgr.R canrep [-h] --af_global AF_GLOBAL --af_keygenes AF_KEYGENES</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="st">                     --batch_name BATCH_NAME [--conda_list CONDA_LIST]</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="st">                     --img_dir IMG_DIR --key_genes KEY_GENES --oncokb_genes</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="st">                     ONCOKB_GENES --somatic_snv_vcf SOMATIC_SNV_VCF</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a><span class="st">                     --somatic_snv_summary SOMATIC_SNV_SUMMARY</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a><span class="st">                     --somatic_sv_tsv SOMATIC_SV_TSV --somatic_sv_vcf</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a><span class="st">                     SOMATIC_SV_VCF --purple_som_gene_cnv PURPLE_SOM_GENE_CNV</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a><span class="st">                     --purple_som_cnv_ann PURPLE_SOM_CNV_ANN --purple_som_cnv</span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a><span class="st">                     PURPLE_SOM_CNV --purple_purity PURPLE_PURITY --purple_qc</span></span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a><span class="st">                     PURPLE_QC --purple_som_snv_vcf PURPLE_SOM_SNV_VCF</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a><span class="st">                     --virusbreakend_tsv VIRUSBREAKEND_TSV --virusbreakend_vcf</span></span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a><span class="st">                     VIRUSBREAKEND_VCF --dragen_hrd DRAGEN_HRD</span></span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a><span class="st">                     --bcftools_stats BCFTOOLS_STATS [--out_file OUT_FILE]</span></span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a><span class="st">                     [--quiet] --result_outdir RESULT_OUTDIR --tumor_name</span></span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a><span class="st">                     TUMOR_NAME</span></span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" tabindex="-1"></a><span class="st">options:</span></span>
<span id="cb4-38"><a href="#cb4-38" tabindex="-1"></a><span class="st">  -h, --help            show this help message and exit</span></span>
<span id="cb4-39"><a href="#cb4-39" tabindex="-1"></a><span class="st">  --af_global AF_GLOBAL</span></span>
<span id="cb4-40"><a href="#cb4-40" tabindex="-1"></a><span class="st">                        Path to `af_tumor.txt` file.</span></span>
<span id="cb4-41"><a href="#cb4-41" tabindex="-1"></a><span class="st">  --af_keygenes AF_KEYGENES</span></span>
<span id="cb4-42"><a href="#cb4-42" tabindex="-1"></a><span class="st">                        Path to `af_tumor_keygenes.txt` file.</span></span>
<span id="cb4-43"><a href="#cb4-43" tabindex="-1"></a><span class="st">  --batch_name BATCH_NAME</span></span>
<span id="cb4-44"><a href="#cb4-44" tabindex="-1"></a><span class="st">                        Name of batch sample.</span></span>
<span id="cb4-45"><a href="#cb4-45" tabindex="-1"></a><span class="st">  --conda_list CONDA_LIST</span></span>
<span id="cb4-46"><a href="#cb4-46" tabindex="-1"></a><span class="st">                        Path to `conda_pkg_list.txt` file.</span></span>
<span id="cb4-47"><a href="#cb4-47" tabindex="-1"></a><span class="st">  --img_dir IMG_DIR     Path to directory containing PURPLE plots.</span></span>
<span id="cb4-48"><a href="#cb4-48" tabindex="-1"></a><span class="st">  --key_genes KEY_GENES</span></span>
<span id="cb4-49"><a href="#cb4-49" tabindex="-1"></a><span class="st">                        Path to UMCCR cancer gene file.</span></span>
<span id="cb4-50"><a href="#cb4-50" tabindex="-1"></a><span class="st">  --oncokb_genes ONCOKB_GENES</span></span>
<span id="cb4-51"><a href="#cb4-51" tabindex="-1"></a><span class="st">                        Path to OncoKB database file.</span></span>
<span id="cb4-52"><a href="#cb4-52" tabindex="-1"></a><span class="st">  --somatic_snv_vcf SOMATIC_SNV_VCF</span></span>
<span id="cb4-53"><a href="#cb4-53" tabindex="-1"></a><span class="st">                        Path to `somatic-PASS.vcf.gz` SNV VCF.</span></span>
<span id="cb4-54"><a href="#cb4-54" tabindex="-1"></a><span class="st">  --somatic_snv_summary SOMATIC_SNV_SUMMARY</span></span>
<span id="cb4-55"><a href="#cb4-55" tabindex="-1"></a><span class="st">                        Path to `somatic_snv_summary.json` JSON.</span></span>
<span id="cb4-56"><a href="#cb4-56" tabindex="-1"></a><span class="st">  --somatic_sv_tsv SOMATIC_SV_TSV</span></span>
<span id="cb4-57"><a href="#cb4-57" tabindex="-1"></a><span class="st">                        Path to `manta.tsv` TSV file.</span></span>
<span id="cb4-58"><a href="#cb4-58" tabindex="-1"></a><span class="st">  --somatic_sv_vcf SOMATIC_SV_VCF</span></span>
<span id="cb4-59"><a href="#cb4-59" tabindex="-1"></a><span class="st">                        Path to `manta.vcf.gz` VCF file.</span></span>
<span id="cb4-60"><a href="#cb4-60" tabindex="-1"></a><span class="st">  --purple_som_gene_cnv PURPLE_SOM_GENE_CNV</span></span>
<span id="cb4-61"><a href="#cb4-61" tabindex="-1"></a><span class="st">                        Path to `purple.cnv.gene.tsv`.</span></span>
<span id="cb4-62"><a href="#cb4-62" tabindex="-1"></a><span class="st">  --purple_som_cnv_ann PURPLE_SOM_CNV_ANN</span></span>
<span id="cb4-63"><a href="#cb4-63" tabindex="-1"></a><span class="st">                        Path to annotated and prioritised</span></span>
<span id="cb4-64"><a href="#cb4-64" tabindex="-1"></a><span class="st">                        `purple.cnv.somatic.tsv`.</span></span>
<span id="cb4-65"><a href="#cb4-65" tabindex="-1"></a><span class="st">  --purple_som_cnv PURPLE_SOM_CNV</span></span>
<span id="cb4-66"><a href="#cb4-66" tabindex="-1"></a><span class="st">                        Path to `purple.cnv.somatic.tsv`.</span></span>
<span id="cb4-67"><a href="#cb4-67" tabindex="-1"></a><span class="st">  --purple_purity PURPLE_PURITY</span></span>
<span id="cb4-68"><a href="#cb4-68" tabindex="-1"></a><span class="st">                        Path to `purple.purity.tsv`.</span></span>
<span id="cb4-69"><a href="#cb4-69" tabindex="-1"></a><span class="st">  --purple_qc PURPLE_QC</span></span>
<span id="cb4-70"><a href="#cb4-70" tabindex="-1"></a><span class="st">                        Path to `purple.qc`.</span></span>
<span id="cb4-71"><a href="#cb4-71" tabindex="-1"></a><span class="st">  --purple_som_snv_vcf PURPLE_SOM_SNV_VCF</span></span>
<span id="cb4-72"><a href="#cb4-72" tabindex="-1"></a><span class="st">                        Path to `purple.somatic.vcf.gz`.</span></span>
<span id="cb4-73"><a href="#cb4-73" tabindex="-1"></a><span class="st">  --virusbreakend_tsv VIRUSBREAKEND_TSV</span></span>
<span id="cb4-74"><a href="#cb4-74" tabindex="-1"></a><span class="st">                        Path to VIRUSBreakend summary file.</span></span>
<span id="cb4-75"><a href="#cb4-75" tabindex="-1"></a><span class="st">  --virusbreakend_vcf VIRUSBREAKEND_VCF</span></span>
<span id="cb4-76"><a href="#cb4-76" tabindex="-1"></a><span class="st">                        Path to VIRUSBreakend VCF file.</span></span>
<span id="cb4-77"><a href="#cb4-77" tabindex="-1"></a><span class="st">  --dragen_hrd DRAGEN_HRD</span></span>
<span id="cb4-78"><a href="#cb4-78" tabindex="-1"></a><span class="st">                        Path to DRAGEN HRD file</span></span>
<span id="cb4-79"><a href="#cb4-79" tabindex="-1"></a><span class="st">  --bcftools_stats BCFTOOLS_STATS</span></span>
<span id="cb4-80"><a href="#cb4-80" tabindex="-1"></a><span class="st">                        Path to bcftools stats file</span></span>
<span id="cb4-81"><a href="#cb4-81" tabindex="-1"></a><span class="st">  --out_file OUT_FILE   Path to output HTML file (needs '</span>.html<span class="st">' suffix) [def:</span></span>
<span id="cb4-82"><a href="#cb4-82" tabindex="-1"></a><span class="st">                        {tumor_name}_cancer_report.html].</span></span>
<span id="cb4-83"><a href="#cb4-83" tabindex="-1"></a><span class="st">  --quiet               Suppress log printing during rendering.</span></span>
<span id="cb4-84"><a href="#cb4-84" tabindex="-1"></a><span class="st">  --result_outdir RESULT_OUTDIR</span></span>
<span id="cb4-85"><a href="#cb4-85" tabindex="-1"></a><span class="st">                        Path to directory to write tidy JSON/TSV results.</span></span>
<span id="cb4-86"><a href="#cb4-86" tabindex="-1"></a><span class="st">  --tumor_name TUMOR_NAME</span></span>
<span id="cb4-87"><a href="#cb4-87" tabindex="-1"></a><span class="st">                        Name of tumor sample.</span></span>
<span id="cb4-88"><a href="#cb4-88" tabindex="-1"></a></span>
<span id="cb4-89"><a href="#cb4-89" tabindex="-1"></a><span class="st">#-----------------------------------#</span></span></code></pre></div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/umccr/gpgr/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/umccr/gpgr/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gpgr</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7502-7545" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
