---
title: "HRDetect"
author: "Peter Diakumis"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette

vignette: >
  %\VignetteIndexEntry{HRDetect}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r knitr_opts, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup, message=FALSE}
require(gpgr)
require(devtools)
require(dplyr)
require(knitr)
```

## Introduction

HRDetect: Homologous Recombination Deficiency Detect
(https://github.com/Nik-Zainal-Group/signature.tools.lib)

## Data

HRDetect takes as input somatic SNV, INDEL, SV and CNV files. We have made it
somewhat easier to feed SNV/INDEL and Manta SV VCFs, together with PURPLE somatic CNV files,
with the `gpgr::hrdetect_run` wrapper.


```{r datafiles}
snvindel_vcf <- system.file("extdata/umccrise/snv/somatic-ensemble-PASS.vcf.gz", package = "gpgr")
sv_vcf <- system.file("extdata/umccrise/sv/manta.vcf.gz", package = "gpgr")
cnv_tsv <- system.file("extdata/purple/purple.cnv.somatic.tsv", package = "gpgr")
```

## Analysis

```{r run_hrdetect}
res <- gpgr::hrdetect_run(
  nm = "sample_A",
  snvindel_vcf = snvindel_vcf,
  sv_vcf = sv_vcf,
  cnv_tsv = cnv_tsv,
  genome = "hg38",
  snvoutdir = tempdir()
)
```

## Results

```{r res}
cbind(var = colnames(res), value = unlist(res[1, ])) |>
  dplyr::as_tibble(.name_repair = "check_unique") |>
  knitr::kable(format = "html", caption = "HRDetect results.")
```

## Session Info

```{r session_info, echo=FALSE}
pkgs_of_interest <- c("base", "gpgr", "signature.tools.lib")
si <- gpgr::session_info_kable(pkgs_of_interest)
si$si_pkg
si$si_pl
```
