---
author: "University of Melbourne Centre for Cancer Research"
date: "`r Sys.time()`"
output:
  html_document:
    theme: cosmo
    css: style.css
    code_download: true
  rmdformats::material:
    highlight: kate
params:
  title: "UMCCR LINX Report"
  sample: "x"
  plot_dir: "x"
  table_dir: "x"
description: "UMCCR LINX Report"
title: "`r paste(params$sample, params$title)`"
---

```{r knitr_opts, include=F}
knitr::opts_chunk$set(
  collapse = TRUE, echo = FALSE,
  warning = FALSE, message = FALSE,
  error = FALSE
)
```

```{r render_report_interactively, echo=FALSE, eval=FALSE}
params <- list(
  sample = "SBJ01196_PRJ211133_L2101598",
  table_dir = here::here("nogit/linx/SBJ01196/tables"),
  plot_dir = here::here("nogit/linx/SBJ01196/plots")
)
render_me <- function() {
  rmarkdown::render(
    here::here("inst/rmd/linx/linx.Rmd"),
    params = params,
    output_file = "SBJ001196.html",
    output_dir = here::here("nogit/linx/test")
  )
}
render_me()
```

```{r load_pkgs, message=FALSE, warning=FALSE}
library(details)
library(dplyr, warn.conflicts = FALSE)
library(DT)
library(glue)
library(gpgr)
library(knitr, include.only = "kable")
library(purrr, include.only = "set_names")
library(stringr, include.only = "str_match")
```

```{r funcs}
blank_lines <- function(n = 10) {
  cat(rep("&nbsp;  ", n), sep = "\n")
}

dt_table <- function(x) {
  x |>
    DT::datatable(
      filter = list(position = "top", clear = FALSE, plain = TRUE),
      class = "cell-border display compact",
      rownames = FALSE,
      extensions = c("Scroller", "Buttons", "KeyTable"),
      options = list(
        scroller = TRUE, scrollY = 400, scrollX = TRUE,
        autoWidth = FALSE, keys = TRUE,
        buttons = c("csv", "excel"), dom = "Blfrtip"
      )
    )
}

linx_path_tab <- function(type) {
  x <- file.path(params$table_dir, glue("{params$sample}.linx.{type}.tsv"))
  stopifnot(file.exists(x))
  x
}

linx_path_plot_chr <- function(chr) {
  pat <- glue("{params$sample}\\.chr{chr}\\.(\\d+)\\.png") |> as.character()
  x <- list.files(file.path(params$plot_dir), pattern = pat, full.names = TRUE)
  stopifnot(length(x) == 1, file.exists(x))
  seg_num <- sub(pat, "\\1", basename(x))
  list(
    path = x,
    segs = seg_num
  )
}

linx_path_plot_cluster <- function() {
  pat <- glue("{params$sample}\\.cluster(\\d+)\\.(.*)\\.sv(\\d+)\\.(\\d+)\\.png") |> as.character()
  x <- list.files(file.path(params$plot_dir), pattern = pat, full.names = TRUE)
  stopifnot(length(x) > 0)
  d <- str_match(x, pat) |>
    dplyr::as_tibble(.name_repair = "minimal") |>
    set_names(c("path", "clusterId", "type", "svId", "nsegs")) |>
    dplyr::mutate(
      path = file.path(params$plot_dir, path),
      nsegs = as.integer(nsegs),
      clusterId = as.integer(clusterId)
    ) |>
    dplyr::arrange(clusterId) |>
    dplyr::mutate(clusterId = sprintf(glue("%0{nchar(max(clusterId))}d"), clusterId))
  d
}
```

## Tables {.tabset #tables}

### Vis {.tabset}

#### Copy Number

- `vis_copy_number.tsv`

```{r vis_copy_number}
linx_path_tab("vis_copy_number") |>
  gpgr::linx_viscopynumber_read() |>
  dt_table()
```

```{r blank1, results='asis'}
blank_lines(2)
```

#### Fusion

- `vis_fusion.tsv`

```{r vis_fusion}
linx_path_tab("vis_fusion") |>
  gpgr::linx_visfusion_read() |>
  dt_table()
```

#### Gene Exon

- `vis_gene_exon.tsv`

```{r vis_gene_exon}
linx_path_tab("vis_gene_exon") |>
  gpgr::linx_visgeneexon_read() |>
  dt_table()
```

#### Protein Domain

- `vis_protein_domain.tsv`

```{r vis_protein_domain}
linx_path_tab("vis_protein_domain") |>
  gpgr::linx_visproteindomain_read() |>
  dt_table()
```

#### Segments

- `vis_segments.tsv`

```{r vis_segments}
linx_path_tab("vis_segments") |>
  gpgr::linx_vissegments_read() |>
  dt_table()
```

#### SV Data

- `vis_sv_data.tsv`

```{r vis_sv_data}
linx_path_tab("vis_sv_data") |>
  gpgr::linx_vissvdata_read() |>
  dt_table()
```

### Main {.tabset}

#### Annotations

- additional annotations of each non PON filtered break junction
- `svs.tsv`

```{r svs}
l <- linx_path_tab("svs") |>
  gpgr::linx_svs_process()

details::details(kable(l$descr), lang = NULL, summary = "Column Description")
dt_table(l$tab)
```

```{r blank2, results='asis'}
blank_lines(2)
```

#### Breakends

- impact of each non PON filtered break junction on each overlapping gene
- `breakend.tsv`

```{r breakend}
l <- linx_path_tab("breakend") |>
  gpgr::linx_breakend_process()
details::details(kable(l$descr), lang = NULL, summary = "Column Description")
dt_table(l$tab)
```

#### Clusters

- clustering of all non PON filtered SV events and their resolved
  classification.
- `clusters.tsv`

```{r clusters}
l <- linx_path_tab("clusters") |>
  gpgr::linx_clusters_process()
details::details(kable(l$descr), lang = NULL, summary = "Column Description")
dt_table(l$tab)
```

#### Links

- segments joining break junction pairs predicted to be linked and phased in
  _cis_ on the derivative chromosome.
- `links.tsv`

```{r links}
l <- linx_path_tab("links") |>
  gpgr::linx_links_process()
details::details(kable(l$descr), lang = NULL, summary = "Column Description")
dt_table(l$tab)
```

#### Fusions

- All inframe and outframe fusions predicted in the sample including HMF
  fusion knowledgebase annotations.
- `fusion.tsv`

```{r fusion}
l <- linx_path_tab("fusion") |>
  gpgr::linx_fusion_process()
details::details(kable(l$descr), lang = NULL, summary = "Column Description")
dt_table(l$tab)
```

#### Driver Catalog

- Reproduction of the driver catalog produced by PURPLE with homozygous
  disruption events appended.
- `driver.catalog.tsv`

```{r drivercatalog}
l <- linx_path_tab("driver.catalog") |>
  gpgr::linx_drivercatalog_process()
details::details(kable(l$descr), lang = NULL, summary = "Column Description")
dt_table(l$tab)
```

#### Drivers

- Linkage of drivers from driver catalog to SV cluster which contributed to
  those drivers including LOH, deletion, disruption and amplification events.
- `drivers.tsv`

```{r drivers}
l <- linx_path_tab("drivers") |>
  gpgr::linx_drivers_process()
details::details(kable(l$descr), lang = NULL, summary = "Column Description")
dt_table(l$tab)
```

## Plots (Chromosome) {.tabset #plots-chrom}

```{r plot_chrom, results='asis'}
for (i in c(1:22, "X")) {
  res <- linx_path_plot_chr(i)
  cat(glue("\n### chr{i}"), "\n")
  cat(glue("\nnsegs: {as.integer(res$segs)}"), "\n\n\n")
  cat(glue("<img src=\"{res$path}\" height=\"1000\"/>"), "\n\n\n")
  blank_lines(1)
  cat("\n\n\n")
}
```

## Plots (Cluster) {.tabset #plots-cluster}

```{r plot_cluster, results='asis'}
cluster_plots <- linx_path_plot_cluster()
for (i in seq_len(nrow(cluster_plots))) {
  r <- cluster_plots[i, ]
  cat(glue("\n### {r$clusterId}"), "\n\n\n")
  cat(glue(
    "clusterId: {r$clusterId} | type: {r$type} | ",
    "svId: {r$svId} | nsegs: {r$nsegs}"
  ), "\n\n\n")
  cat(glue("<img src=\"{r$path}\" height=\"1000\"/>"), "\n\n\n")
  blank_lines(2)
  cat("\n\n\n")
}
```

## Addendum {.tabset #addendum}

<details>
<summary>Show</summary>


### Session Info

```{r session_info, echo=FALSE}
pkgs_of_interest <- c("base", "gpgr")
si <- gpgr::session_info_kable(pkgs_of_interest)
si$si_pkg
si$si_pl
```

### Report Inputs

```{r report_inputs}
report_inputs <- dplyr::tibble(key = names(params), value = unlist(params))
report_inputs |> kable()
```

</details>
