---
author: "University of Melbourne Centre for Cancer Research"
date: "`r Sys.time()`"
output:
  html_document:
    theme: cosmo
    css: style.css
    toc: true
    code_download: true
  rmdformats::material:
    highlight: kate
params:
  title: "UMCCR LINX Report"
  sample: "x"
  plot_dir: "x"
  table_dir: "x"
description: "UMCCR LINX Report"
title: "`r paste(params$sample, params$title)`"
---

```{r knitr_opts, include=F}
knitr::opts_chunk$set(
  collapse = TRUE, echo = FALSE,
  warning = FALSE, message = FALSE,
  error = TRUE
) # continue running report even after chunk errors
```

```{r render_report_interactively, echo=F, eval=FALSE}
params <- list(
  sample = "SEQC-II_Tumor_50pc",
  table_dir = here::here("nogit/linx/SEQCII/GPL_shortcut/tables"),
  plot_dir = here::here("inst/linx/SEQCII/GPL_shortcut/plots")
)
render_me <- function() {
  rmarkdown::render(
    here::here("inst/rmd/linx/linx.Rmd"),
    params = params
  )
}
render_me()
```

```{r load_pkgs}
# main
library(gpgr)
library(tidyverse)
# other
library(DT)
library(here)
library(glue)
```

```{r funcs}
dt_table <- function(x) {
  x |>
    DT::datatable(
      filter = list(position = "top", clear = FALSE, plain = TRUE),
      class = "cell-border display compact",
      rownames = FALSE,
      extensions = c("Scroller", "Buttons", "KeyTable"),
      options = list(
        scroller = TRUE, scrollY = 400, scrollX = TRUE,
        autoWidth = FALSE, keys = TRUE,
        buttons = c("csv", "excel"), dom = "Blfrtip"
      )
    )
}

linx_tab_path <- function(type) {
  file.path(params$table_dir, glue("{params$sample}.linx.{type}.tsv"))
}
```

## Tables Main {.tabset #tables-main}

### Annotations

- additional annotations of each non PON filtered break junction
- `svs.tsv`

```{r}
linx_tab_path("svs") |>
  gpgr::linx_svs_read() |>
  dt_table()
```


### Breakends

- impact of each non PON filtered break junction on each overlapping gene
- `breakend.tsv`

```{r}
linx_tab_path("breakend") |>
  gpgr::linx_breakend_read() |>
  dt_table()
```

### Clusters

- clustering of all non PON filtered SV events and their resolved
  classification.
- `clusters.tsv`

```{r}
linx_tab_path("clusters") |>
  gpgr::linx_clusters_read() |>
  dt_table()
```

### Links

- segments joining break junction pairs predicted to be linked and phased in
  _cis_ on the derivative chromosome.
- `links.tsv`

```{r}
linx_tab_path("links") |>
  gpgr::linx_links_read() |>
  dt_table()
```

### Fusions

- All inframe and outframe fusions predicted in the sample including HMF
  fusion knowledgebase annotations.
- `fusion.tsv`

```{r}
linx_tab_path("fusion") |>
  gpgr::linx_fusion_read() |>
  dt_table()
```

### Driver Catalog

- Reproduction of the driver catalog produced by PURPLE with homozygous
  disruption events appended.
- `driver.catalog.tsv`

```{r}
linx_tab_path("driver.catalog") |>
  gpgr::linx_drivercatalog_read() |>
  dt_table()
```

### Drivers

- Linkage of drivers from driver catalog to SV cluster which contributed to
  those drivers including LOH, deletion, disruption and amplification events.
- `drivers.tsv`

```{r}
linx_tab_path("drivers") |>
  gpgr::linx_drivers_read() |>
  dt_table()
```

## Tables Vis {.tabset #tables-vis}

### Copy Number

- `vis_copy_number.tsv`

```{r}
linx_tab_path("vis_copy_number") |>
  gpgr::linx_viscopynumber_read() |>
  dt_table()
```

### Fusion

- `vis_fusion.tsv`

```{r}
linx_tab_path("vis_fusion") |>
  gpgr::linx_visfusion_read() |>
  dt_table()
```

### Gene Exon

- `vis_gene_exon.tsv`

```{r}
linx_tab_path("vis_gene_exon") |>
  gpgr::linx_visgeneexon_read() |>
  dt_table()
```

### Protein Domain

- `vis_protein_domain.tsv`

```{r}
linx_tab_path("vis_protein_domain") |>
  gpgr::linx_visproteindomain_read() |>
  dt_table()
```

### Segments

- `vis_segments.tsv`

```{r}
linx_tab_path("vis_segments") |>
  gpgr::linx_vissegments_read() |>
  dt_table()
```

### SV Data

- `vis_sv_data.tsv`

```{r}
linx_tab_path("vis_sv_data") |>
  gpgr::linx_vissvdata_read() |>
  dt_table()
```

## Plots (Chromosome) {.tabset #plots-chrom}

## Plots (Cluster) {.tabset #plots-cluster}

## Addendum {.tabset #addendum}

<details>
<summary>Show</summary>


### Session Info

```{r session_info, echo=FALSE}
pkgs_of_interest <- c("base", "gpgr", "tidyverse")
si <- gpgr::session_info_kable(pkgs_of_interest)
si$si_pkg
si$si_pl
```

### Report Inputs

```{r report_inputs}
report_inputs <- dplyr::tibble(key = names(params), value = unlist(params))
report_inputs |> knitr::kable()
```

</details>
