# Generated by roxytest: Do not edit by hand!

# File R/hrdetect.R: @testexamples

test_that("Function hrdetect_read_snvindel_vcf() @ L20", {
  
  x <- system.file("extdata/umccrise/snv/somatic-ensemble-PASS.vcf.gz", package = "gpgr")
  (l <- hrdetect_read_snvindel_vcf(x))
  expect_equal(length(l), 2)
  expect_equal(names(l), c("snv", "indel"))
  expect_equal(colnames(l$snv), c("chr", "position", "REF", "ALT"))
  expect_equal(colnames(l$indel), c("chr", "position", "REF", "ALT"))
})


test_that("Function hrdetect_read_sv_vcf() @ L72", {
  
  x <- system.file("extdata/umccrise/sv/manta.vcf.gz", package = "gpgr")
  sv_bedpe <- hrdetect_read_sv_vcf(x, nm = "SAMPLE")
  head(sv_bedpe)
  expect_equal(nrow(sv_bedpe), 190)
  expect_equal(colnames(sv_bedpe), c("chrom1", "start1", "end1", "chrom2",
               "start2", "end2", "sample", "strand1", "strand2"))
})


test_that("Function hrdetect_read_purple_cnv() @ L114", {
  
  x <- system.file("extdata/purple/purple.cnv.somatic.tsv", package = "gpgr")
  (cnv <- hrdetect_read_purple_cnv(x))
  expect_equal(colnames(cnv), c("Chromosome", "chromStart", "chromEnd",
                                "total.copy.number.inTumour",
                                "minor.copy.number.inTumour"))
})


test_that("Function hrdetect_prep_snvindel() @ L155", {
  
  x <- system.file("extdata/umccrise/snv/somatic-ensemble-PASS.vcf.gz", package = "gpgr")
  (l <- hrdetect_prep_snvindel(x, nm = "sampleA", outdir = tempdir()))
  expect_equal(c("snv_results", "indel_results"), names(l))
  expect_equal(c("sig", "exposure", "pvalue"), colnames(l[["snv_results"]]))
  expect_equal(colnames(l[["indel_results"]])[c(1, 7)], c("sample", "del.mh.prop"))
})


test_that("Function hrdetect_prep_sv() @ L230", {
  
  x <- system.file("extdata/umccrise/sv/manta.vcf.gz", package = "gpgr")
  nm <- "SampleA"
  (d <- hrdetect_prep_sv(x, nm))
  expect_equal(colnames(d), nm)
  expect_true(inherits(d, "data.frame"))
})


test_that("Function hrdetect_prep_cnv() @ L258", {
  
  x <- system.file("extdata/purple/purple.cnv.somatic.tsv", package = "gpgr")
  (l <- hrdetect_prep_cnv(x, nm = "SampleA"))
  expect_equal(colnames(l), c("name", "hrdloh_index"))
  expect_equal(nrow(l), 1)
})


test_that("Function hrdetect_run() @ L305", {
  
  snvindel_vcf <- system.file(
    "extdata/umccrise/snv/somatic-ensemble-PASS.vcf.gz",
    package = "gpgr"
  )
  sv_vcf <- system.file("extdata/umccrise/sv/manta.vcf.gz", package = "gpgr")
  cnv_tsv <- system.file("extdata/purple/purple.cnv.somatic.tsv", package = "gpgr")
  nm <- "SampleA"
  genome <- "hg38"
  snvoutdir <- tempdir()
  (res <- hrdetect_run(nm, snvindel_vcf, sv_vcf, cnv_tsv, genome, snvoutdir))
  # hrdetect_run(nm, snvindel_vcf, sv_vcf, cnv_tsv, genome, snvoutdir,
  #              outpath = "nogit/hrdetect_results.json.gz")
  expect_equal(colnames(res), c("sample", "Probability", "intercept", "del.mh.prop", "SNV3",
                                "SV3", "SV5", "hrdloh_index", "SNV8"))
  expect_true(inherits(res, "data.frame"))
})

