<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gpgr">
<title>PURPLE • gpgr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PURPLE">
<meta property="og:description" content="gpgr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gpgr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.2.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/devnotes.html">Developers</a>
    <a class="dropdown-item" href="../articles/purple.html">PURPLE</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/umccr/gpgr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>PURPLE</h1>
            
            <h4 data-toc-skip class="date">2025-09-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/umccr/gpgr/blob/HEAD/vignettes/purple.Rmd" class="external-link"><code>vignettes/purple.Rmd</code></a></small>
      <div class="d-none name"><code>purple.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://umccr.github.io/gpgr/">gpgr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/" class="external-link">knitr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>PURPLE: Purity Ploidy Estimator (<a href="https://github.com/hartwigmedical/hmftools/tree/master/purple" class="external-link uri">https://github.com/hartwigmedical/hmftools/tree/master/purple</a>).</p>
<p>PURPLE combines B-allele frequency, read depth ratios, small variants
and structural variants to estimate the purity and copy number profile
of a tumor sample.</p>
<p>It outputs several files, some of which are displayed below.</p>
</div>
<div class="section level2">
<h2 id="data-munging">Data Munging<a class="anchor" aria-label="anchor" href="#data-munging"></a>
</h2>
<div class="section level3">
<h3 id="somatic-cnvs-per-chromosome">Somatic CNVs (per chromosome)<a class="anchor" aria-label="anchor" href="#somatic-cnvs-per-chromosome"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnv_som</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.cnv.somatic.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_cnv_som_process.html">purple_cnv_som_process</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Description
</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnv_som</span><span class="op">$</span><span class="va">descr</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per chromosome) Columns."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>PURPLE Somatic CNVs (per chromosome) Columns.</caption>
<colgroup>
<col width="6%">
<col width="93%">
</colgroup>
<thead><tr>
<th align="left">Column</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr>
<td align="left">Chr/Start/End</td>
<td align="left">Coordinates of copy number segment</td>
</tr>
<tr>
<td align="left">CN</td>
<td align="left">Fitted absolute copy number of segment adjusted for
purity and ploidy</td>
</tr>
<tr>
<td align="left">CN Min+Maj</td>
<td align="left">CopyNumber of minor + major allele adjusted for
purity</td>
</tr>
<tr>
<td align="left">Start/End SegSupport</td>
<td align="left">Type of SV support for the CN breakpoint at start/end
of region. Allowed values: CENTROMERE, TELOMERE, INV, DEL, DUP, BND
(translocation), SGL (single breakend SV support), NONE (no SV support
for CN breakpoint), MULT (multiple SV support at exact breakpoint)</td>
</tr>
<tr>
<td align="left">Method</td>
<td align="left">Method used to determine the CN of the region. Allowed
values: BAF_WEIGHTED (avg of all depth windows for the region),
STRUCTURAL_VARIANT (inferred using ploidy of flanking SVs), LONG_ARM
(inferred from the long arm), GERMLINE_AMPLIFICATION (inferred using
special logic to handle regions of germline amplification)</td>
</tr>
<tr>
<td align="left">BAF (count)</td>
<td align="left">Tumor BAF after adjusted for purity and ploidy (Count
of AMBER baf points covered by this segment)</td>
</tr>
<tr>
<td align="left">GC (windowCount)</td>
<td align="left">Proportion of segment that is G or C (Count of COBALT
windows covered by this segment)</td>
</tr>
</tbody>
</table></details><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnv_som</span><span class="op">$</span><span class="va">tab</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per chromosome) Summary Table."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>PURPLE Somatic CNVs (per chromosome) Summary Table.</caption>
<colgroup>
<col width="4%">
<col width="9%">
<col width="9%">
<col width="3%">
<col width="10%">
<col width="19%">
<col width="17%">
<col width="11%">
<col width="15%">
</colgroup>
<thead><tr>
<th align="left">Chr</th>
<th align="right">Start</th>
<th align="right">End</th>
<th align="right">CN</th>
<th align="left">CN Min+Maj</th>
<th align="left">Start/End SegSupport</th>
<th align="left">Method</th>
<th align="left">BAF (count)</th>
<th align="left">GC (windowCount)</th>
</tr></thead>
<tbody>
<tr>
<td align="left">chr1</td>
<td align="right">1</td>
<td align="right">123605522</td>
<td align="right">1.0</td>
<td align="left">0+1</td>
<td align="left">TELOMERE-CENTROMERE</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.98 (20830)</td>
<td align="left">0.42 (107822)</td>
</tr>
<tr>
<td align="left">chr1</td>
<td align="right">123605523</td>
<td align="right">200044314</td>
<td align="right">2.0</td>
<td align="left">1+1</td>
<td align="left">CENTROMERE-DUP</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.5 (10239)</td>
<td align="left">0.4 (47332)</td>
</tr>
<tr>
<td align="left">chr1</td>
<td align="right">200044315</td>
<td align="right">200044570</td>
<td align="right">2.8</td>
<td align="left">1+1.8</td>
<td align="left">DUP-DUP</td>
<td align="left">STRUCTURAL_VARIANT</td>
<td align="left">0.65 (0)</td>
<td align="left">0 (0)</td>
</tr>
<tr>
<td align="left">chr1</td>
<td align="right">200044571</td>
<td align="right">248956422</td>
<td align="right">2.0</td>
<td align="left">1+1</td>
<td align="left">DUP-TELOMERE</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.5 (10341)</td>
<td align="left">0.42 (43373)</td>
</tr>
<tr>
<td align="left">chr2</td>
<td align="right">1</td>
<td align="right">93139350</td>
<td align="right">2.0</td>
<td align="left">1+1</td>
<td align="left">TELOMERE-CENTROMERE</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.51 (19339)</td>
<td align="left">0.41 (81949)</td>
</tr>
<tr>
<td align="left">chr2</td>
<td align="right">93139351</td>
<td align="right">219955359</td>
<td align="right">2.0</td>
<td align="left">1+1</td>
<td align="left">CENTROMERE-BND</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.5 (19047)</td>
<td align="left">0.39 (112493)</td>
</tr>
<tr>
<td align="left">chr2</td>
<td align="right">219955360</td>
<td align="right">225225069</td>
<td align="right">1.0</td>
<td align="left">0+1</td>
<td align="left">BND-BND</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.98 (1284)</td>
<td align="left">0.4 (5069)</td>
</tr>
<tr>
<td align="left">chr2</td>
<td align="right">225225070</td>
<td align="right">242193529</td>
<td align="right">2.0</td>
<td align="left">1+1</td>
<td align="left">BND-TELOMERE</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.51 (4236)</td>
<td align="left">0.44 (15099)</td>
</tr>
<tr>
<td align="left">chr3</td>
<td align="right">1</td>
<td align="right">92214015</td>
<td align="right">1.0</td>
<td align="left">0+1</td>
<td align="left">TELOMERE-CENTROMERE</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.98 (18629)</td>
<td align="left">0.4 (83984)</td>
</tr>
<tr>
<td align="left">chr3</td>
<td align="right">92214016</td>
<td align="right">198295559</td>
<td align="right">2.0</td>
<td align="left">1+1</td>
<td align="left">CENTROMERE-TELOMERE</td>
<td align="left">BAF_WEIGHTED</td>
<td align="left">0.5 (21114)</td>
<td align="left">0.39 (95873)</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="somatic-cnvs-per-gene">Somatic CNVs (per gene)<a class="anchor" aria-label="anchor" href="#somatic-cnvs-per-gene"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">umccr_key_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/ref/somatic_panel-v24.03.0.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span></span>
<span><span class="va">cnv_som_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.cnv.gene.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_cnv_som_gene_process.html">purple_cnv_som_gene_process</a></span><span class="op">(</span>g <span class="op">=</span> <span class="va">umccr_key_genes</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Description
</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnv_som_gene</span><span class="op">$</span><span class="va">descr</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per gene) Columns."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>PURPLE Somatic CNVs (per gene) Columns.</caption>
<colgroup>
<col width="21%">
<col width="78%">
</colgroup>
<thead><tr>
<th align="left">Column</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr>
<td align="left">gene</td>
<td align="left">Name of gene</td>
</tr>
<tr>
<td align="left">minCN/maxCN</td>
<td align="left">Min/Max copy number found in gene exons</td>
</tr>
<tr>
<td align="left">chrom/start/end</td>
<td align="left">Chromosome/start/end location of gene transcript</td>
</tr>
<tr>
<td align="left">chrBand</td>
<td align="left">Chromosome band of the gene</td>
</tr>
<tr>
<td align="left">onco_or_ts</td>
<td align="left">oncogene (‘oncogene’), tumor suppressor (‘tsgene’), or
both (‘onco+ts’), as reported by <a href="https://github.com/jakelever/cancermine" class="external-link">Cancermine</a>
</td>
</tr>
<tr>
<td align="left">transcriptID</td>
<td align="left">Ensembl transcript ID (dot version)</td>
</tr>
<tr>
<td align="left">minMinorAlleleCN</td>
<td align="left">Minimum allele ploidy found over the gene exons -
useful for identifying LOH events</td>
</tr>
<tr>
<td align="left">somReg (somaticRegions)</td>
<td align="left">Count of somatic copy number regions this gene
spans</td>
</tr>
<tr>
<td align="left">minReg (minRegions)</td>
<td align="left">Number of somatic regions inside the gene that share
the min copy number</td>
</tr>
<tr>
<td align="left">minRegStartEnd</td>
<td align="left">Start/End base of the copy number region overlapping
the gene with the minimum copy number</td>
</tr>
<tr>
<td align="left">minRegSupportStartEndMethod</td>
<td align="left">Start/end support of the CN region overlapping the gene
with the min CN (plus determination method)</td>
</tr>
</tbody>
</table></details><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnv_som_gene</span><span class="op">$</span><span class="va">tab</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per gene) Summary Table."</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>PURPLE Somatic CNVs (per gene) Summary Table.</caption>
<colgroup>
<col width="4%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="7%">
<col width="10%">
<col width="11%">
<col width="4%">
<col width="4%">
<col width="9%">
<col width="18%">
</colgroup>
<thead><tr>
<th align="left">gene</th>
<th align="right">minCN</th>
<th align="right">maxCN</th>
<th align="left">chrom</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="left">chrBand</th>
<th align="left">onco_or_ts</th>
<th align="left">transcriptID</th>
<th align="right">minMinorAlleleCN</th>
<th align="right">somReg</th>
<th align="right">minReg</th>
<th align="left">minRegStartEnd</th>
<th align="left">minRegSupportStartEndMethod</th>
</tr></thead>
<tbody>
<tr>
<td align="left">SDHA</td>
<td align="right">5.6640</td>
<td align="right">5.6640</td>
<td align="left">chr5</td>
<td align="right">218303</td>
<td align="right">257082</td>
<td align="left">p15.33</td>
<td align="left">tsgene</td>
<td align="left">ENST00000264932</td>
<td align="right">1.8901</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">89179-297781</td>
<td align="left">DEL-DEL (BAF_WEIGHTED)</td>
</tr>
<tr>
<td align="left">DUSP22</td>
<td align="right">4.6638</td>
<td align="right">4.6638</td>
<td align="left">chr6</td>
<td align="right">291630</td>
<td align="right">351355</td>
<td align="left">p25.3</td>
<td align="left">tsgene</td>
<td align="left">ENST00000419235</td>
<td align="right">1.6754</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">1-834611</td>
<td align="left">TELOMERE-DEL (BAF_WEIGHTED)</td>
</tr>
<tr>
<td align="left">IRF4</td>
<td align="right">4.6638</td>
<td align="right">4.6638</td>
<td align="left">chr6</td>
<td align="right">391752</td>
<td align="right">411443</td>
<td align="left">p25.3</td>
<td align="left">oncogene</td>
<td align="left">ENST00000380956</td>
<td align="right">1.6754</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">1-834611</td>
<td align="left">TELOMERE-DEL (BAF_WEIGHTED)</td>
</tr>
<tr>
<td align="left">KDM5A</td>
<td align="right">4.2953</td>
<td align="right">4.2953</td>
<td align="left">chr12</td>
<td align="right">280057</td>
<td align="right">389320</td>
<td align="left">p13.33</td>
<td align="left">oncogene</td>
<td align="left">ENST00000399788</td>
<td align="right">0.0000</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">1-1210023</td>
<td align="left">TELOMERE-DEL (BAF_WEIGHTED)</td>
</tr>
<tr>
<td align="left">CRLF2</td>
<td align="right">3.7271</td>
<td align="right">3.7271</td>
<td align="left">chrX</td>
<td align="right">1187549</td>
<td align="right">1212723</td>
<td align="left">p22.33</td>
<td align="left">oncogene</td>
<td align="left">ENST00000400841</td>
<td align="right">0.0000</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">924013-2677577</td>
<td align="left">DUP-DUP (BAF_WEIGHTED)</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="purity">Purity<a class="anchor" aria-label="anchor" href="#purity"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">purity</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.purity.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_purity_read.html">purple_purity_read</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">purity</span><span class="op">$</span><span class="va">summary</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"PURPLE Purity Summary Table."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>PURPLE Purity Summary Table.</caption>
<colgroup>
<col width="2%">
<col width="10%">
<col width="11%">
<col width="76%">
</colgroup>
<thead><tr>
<th align="right">n</th>
<th align="left">variable</th>
<th align="left">value</th>
<th align="left">details</th>
</tr></thead>
<tbody>
<tr>
<td align="right">2</td>
<td align="left">Purity</td>
<td align="left">0.99 (0.98-1)</td>
<td align="left">Purity of tumor in the sample (and min-max with score
within 10% of best).</td>
</tr>
<tr>
<td align="right">3</td>
<td align="left">Ploidy</td>
<td align="left">2.86 (2.78-2.94)</td>
<td align="left">Average ploidy of tumor sample after adjusting for
purity (and min-max with score within 10% of best).</td>
</tr>
<tr>
<td align="right">4</td>
<td align="left">Gender</td>
<td align="left">MALE</td>
<td align="left">Gender as inferred by AMBER/COBALT.</td>
</tr>
<tr>
<td align="right">7</td>
<td align="left">WGD</td>
<td align="left">TRUE</td>
<td align="left">Whole genome duplication (more than 10 autosomes have
average major allele ploidy &gt; 1.5).</td>
</tr>
<tr>
<td align="right">8</td>
<td align="left">PolyclonalProp</td>
<td align="left">0.13</td>
<td align="left">Proportion of CN regions that are more than 0.25 from a
whole CN</td>
</tr>
<tr>
<td align="right">9</td>
<td align="left">DiploidyProp</td>
<td align="left">0.03 (0.02-0.04)</td>
<td align="left">Proportion of CN regions that have 1 (+- 0.2) minor and
major allele.</td>
</tr>
<tr>
<td align="right">10</td>
<td align="left">TMB</td>
<td align="left">15.13 (HIGH)</td>
<td align="left">Tumor mutational burden (# PASS variants per Megabase)
(Status: ‘HIGH’ (&gt;10 PASS per Mb), ‘LOW’ or ‘UNKNOWN’).</td>
</tr>
<tr>
<td align="right">11</td>
<td align="left">TML</td>
<td align="left">349 (HIGH)</td>
<td align="left">Tumor mutational load (# of missense variants) (Status:
‘HIGH’, ‘LOW’ or ‘UNKNOWN’).</td>
</tr>
<tr>
<td align="right">12</td>
<td align="left">TMB-SV</td>
<td align="left">1267</td>
<td align="left"># of non inferred, non single passing SVs.</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="qc">QC<a class="anchor" aria-label="anchor" href="#qc"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.qc"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_qc_read.html">purple_qc_read</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">qc</span><span class="op">$</span><span class="va">summary</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"PURPLE QC Summary Table."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>PURPLE QC Summary Table.</caption>
<colgroup>
<col width="2%">
<col width="16%">
<col width="21%">
<col width="60%">
</colgroup>
<thead><tr>
<th align="right">n</th>
<th align="left">variable</th>
<th align="left">value</th>
<th align="left">details</th>
</tr></thead>
<tbody>
<tr>
<td align="right">1</td>
<td align="left">QC_Status</td>
<td align="left">FAIL_CONTAMINATION</td>
<td align="left">See ‘Description’.</td>
</tr>
<tr>
<td align="right">13</td>
<td align="left">Method</td>
<td align="left">NORMAL</td>
<td align="left">Fit method (NORMAL, HIGHLY_DIPLOID, SOMATIC or
NO_TUMOR).</td>
</tr>
<tr>
<td align="right">14</td>
<td align="left">CopyNumberSegments</td>
<td align="left">1428 (Unsupported: 2)</td>
<td align="left"># of CN segments.</td>
</tr>
<tr>
<td align="right">2</td>
<td align="left">Purity</td>
<td align="left">0.8600</td>
<td align="left"></td>
</tr>
<tr>
<td align="right">17</td>
<td align="left">Gender</td>
<td align="left">Amber: MALE; Cobalt: MALE</td>
<td align="left"></td>
</tr>
<tr>
<td align="right">14</td>
<td align="left">DeletedGenes</td>
<td align="left">2</td>
<td align="left"># of homozygously deleted genes.</td>
</tr>
<tr>
<td align="right">15</td>
<td align="left">Contamination</td>
<td align="left">0.05</td>
<td align="left">Rate of contamination in tumor sample as determined by
AMBER.</td>
</tr>
<tr>
<td align="right">16</td>
<td align="left">GermlineAberrations</td>
<td align="left">NONE</td>
<td align="left">Can be one or more of: KLINEFELTER,
TRISOMY_X/21/13/18/15, XYY, MOSAIC_X.</td>
</tr>
<tr>
<td align="right">18</td>
<td align="left">AmberMeanDepth</td>
<td align="left">30</td>
<td align="left">Mean depth as determined by AMBER.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<table class="table">
<caption>Main packages used in this vignette.</caption>
<thead><tr>
<th align="left">package</th>
<th align="left">version</th>
<th align="left">datestamp</th>
<th align="left">source</th>
</tr></thead>
<tbody>
<tr>
<td align="left">base</td>
<td align="left">4.2.3</td>
<td align="left">2024-06-16</td>
<td align="left">local</td>
</tr>
<tr>
<td align="left">gpgr</td>
<td align="left">2.2.4</td>
<td align="left">2025-09-24</td>
<td align="left">local</td>
</tr>
</tbody>
</table>
<table class="table">
<caption>Platform information.</caption>
<colgroup>
<col width="11%">
<col width="88%">
</colgroup>
<thead><tr>
<th align="left">name</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr>
<td align="left">version</td>
<td align="left">R version 4.2.3 (2023-03-15)</td>
</tr>
<tr>
<td align="left">os</td>
<td align="left">Ubuntu 24.04.3 LTS</td>
</tr>
<tr>
<td align="left">system</td>
<td align="left">x86_64, linux-gnu</td>
</tr>
<tr>
<td align="left">ui</td>
<td align="left">X11</td>
</tr>
<tr>
<td align="left">language</td>
<td align="left">en</td>
</tr>
<tr>
<td align="left">collate</td>
<td align="left">C.UTF-8</td>
</tr>
<tr>
<td align="left">ctype</td>
<td align="left">C.UTF-8</td>
</tr>
<tr>
<td align="left">tz</td>
<td align="left">Etc/UTC</td>
</tr>
<tr>
<td align="left">date</td>
<td align="left">2025-09-24</td>
</tr>
<tr>
<td align="left">pandoc</td>
<td align="left">3.8 @ /home/runner/micromamba/envs/pkgdownenv/bin/ (via
rmarkdown)</td>
</tr>
</tbody>
</table>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
