<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gpgr">
<title>PURPLE â€¢ gpgr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PURPLE">
<meta property="og:description" content="gpgr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gpgr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/devnotes.html">Developers</a>
    <a class="dropdown-item" href="../articles/linx.html">LINX</a>
    <a class="dropdown-item" href="../articles/purple.html">PURPLE</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/umccr/gpgr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>PURPLE</h1>
            
            <h4 data-toc-skip class="date">2022-02-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/umccr/gpgr/blob/HEAD/vignettes/purple.Rmd" class="external-link"><code>vignettes/purple.Rmd</code></a></small>
      <div class="d-none name"><code>purple.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://umccr.github.io/gpgr/">gpgr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/" class="external-link">knitr</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>PURPLE: Purity Ploidy Estimator (<a href="https://github.com/hartwigmedical/hmftools/tree/master/purple" class="external-link uri">https://github.com/hartwigmedical/hmftools/tree/master/purple</a>).</p>
<p>PURPLE combines B-allele frequency, read depth ratios, small variants
and structural variants to estimate the purity and copy number profile
of a tumor sample.</p>
<p>It outputs several files, some of which are displayed below.</p>
</div>
<div class="section level2">
<h2 id="data-munging">Data Munging<a class="anchor" aria-label="anchor" href="#data-munging"></a>
</h2>
<div class="section level3">
<h3 id="somatic-cnvs-per-chromosome">Somatic CNVs (per chromosome)<a class="anchor" aria-label="anchor" href="#somatic-cnvs-per-chromosome"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_som</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.cnv.somatic.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> |&gt;
  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_cnv_som_process.html">purple_cnv_som_process</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<details><summary>
Description
</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_som</span><span class="op">$</span><span class="va">descr</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per chromosome) Columns."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Somatic CNVs (per chromosome) Columns.
</caption>
<thead><tr>
<th style="text-align:left;">
Column
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Chr/Start/End
</td>
<td style="text-align:left;">
Coordinates of copy number segment
</td>
</tr>
<tr>
<td style="text-align:left;">
CN
</td>
<td style="text-align:left;">
Fitted absolute copy number of segment adjusted for purity and ploidy
</td>
</tr>
<tr>
<td style="text-align:left;">
CN Min+Maj
</td>
<td style="text-align:left;">
CopyNumber of minor + major allele adjusted for purity
</td>
</tr>
<tr>
<td style="text-align:left;">
Start/End SegSupport
</td>
<td style="text-align:left;">
Type of SV support for the CN breakpoint at start/end of region. Allowed
values: CENTROMERE, TELOMERE, INV, DEL, DUP, BND (translocation), SGL
(single breakend SV support), NONE (no SV support for CN breakpoint),
MULT (multiple SV support at exact breakpoint)
</td>
</tr>
<tr>
<td style="text-align:left;">
Method
</td>
<td style="text-align:left;">
Method used to determine the CN of the region. Allowed values:
BAF_WEIGHTED (avg of all depth windows for the region),
STRUCTURAL_VARIANT (inferred using ploidy of flanking SVs), LONG_ARM
(inferred from the long arm), GERMLINE_AMPLIFICATION (inferred using
special logic to handle regions of germline amplification)
</td>
</tr>
<tr>
<td style="text-align:left;">
BAF (count)
</td>
<td style="text-align:left;">
Tumor BAF after adjusted for purity and ploidy (Count of AMBER baf
points covered by this segment)
</td>
</tr>
<tr>
<td style="text-align:left;">
GC (windowCount)
</td>
<td style="text-align:left;">
Proportion of segment that is G or C (Count of COBALT windows covered by
this segment)
</td>
</tr>
</tbody>
</table></details><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_som</span><span class="op">$</span><span class="va">tab</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per chromosome) Summary Table."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Somatic CNVs (per chromosome) Summary Table.
</caption>
<thead><tr>
<th style="text-align:left;">
Chr
</th>
<th style="text-align:right;">
Start
</th>
<th style="text-align:right;">
End
</th>
<th style="text-align:right;">
CN
</th>
<th style="text-align:left;">
CN Min+Maj
</th>
<th style="text-align:left;">
Start/End SegSupport
</th>
<th style="text-align:left;">
Method
</th>
<th style="text-align:left;">
BAF (count)
</th>
<th style="text-align:left;">
GC (windowCount)
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
123605522
</td>
<td style="text-align:right;">
1.0
</td>
<td style="text-align:left;">
0+1
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.98 (20830)
</td>
<td style="text-align:left;">
0.42 (107822)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
123605523
</td>
<td style="text-align:right;">
200044314
</td>
<td style="text-align:right;">
2.0
</td>
<td style="text-align:left;">
1+1
</td>
<td style="text-align:left;">
CENTROMERE-DUP
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.5 (10239)
</td>
<td style="text-align:left;">
0.4 (47332)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
200044315
</td>
<td style="text-align:right;">
200044570
</td>
<td style="text-align:right;">
2.8
</td>
<td style="text-align:left;">
1+1.8
</td>
<td style="text-align:left;">
DUP-DUP
</td>
<td style="text-align:left;">
STRUCTURAL_VARIANT
</td>
<td style="text-align:left;">
0.65 (0)
</td>
<td style="text-align:left;">
0 (0)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
200044571
</td>
<td style="text-align:right;">
248956422
</td>
<td style="text-align:right;">
2.0
</td>
<td style="text-align:left;">
1+1
</td>
<td style="text-align:left;">
DUP-TELOMERE
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.5 (10341)
</td>
<td style="text-align:left;">
0.42 (43373)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
93139350
</td>
<td style="text-align:right;">
2.0
</td>
<td style="text-align:left;">
1+1
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.51 (19339)
</td>
<td style="text-align:left;">
0.41 (81949)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
93139351
</td>
<td style="text-align:right;">
219955359
</td>
<td style="text-align:right;">
2.0
</td>
<td style="text-align:left;">
1+1
</td>
<td style="text-align:left;">
CENTROMERE-BND
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.5 (19047)
</td>
<td style="text-align:left;">
0.39 (112493)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
219955360
</td>
<td style="text-align:right;">
225225069
</td>
<td style="text-align:right;">
1.0
</td>
<td style="text-align:left;">
0+1
</td>
<td style="text-align:left;">
BND-BND
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.98 (1284)
</td>
<td style="text-align:left;">
0.4 (5069)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
225225070
</td>
<td style="text-align:right;">
242193529
</td>
<td style="text-align:right;">
2.0
</td>
<td style="text-align:left;">
1+1
</td>
<td style="text-align:left;">
BND-TELOMERE
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.51 (4236)
</td>
<td style="text-align:left;">
0.44 (15099)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
92214015
</td>
<td style="text-align:right;">
1.0
</td>
<td style="text-align:left;">
0+1
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.98 (18629)
</td>
<td style="text-align:left;">
0.4 (83984)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
92214016
</td>
<td style="text-align:right;">
198295559
</td>
<td style="text-align:right;">
2.0
</td>
<td style="text-align:left;">
1+1
</td>
<td style="text-align:left;">
CENTROMERE-TELOMERE
</td>
<td style="text-align:left;">
BAF_WEIGHTED
</td>
<td style="text-align:left;">
0.5 (21114)
</td>
<td style="text-align:left;">
0.39 (95873)
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="somatic-cnvs-per-gene">Somatic CNVs (per gene)<a class="anchor" aria-label="anchor" href="#somatic-cnvs-per-gene"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">umccr_key_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/ref/umccr_cancer_genes_2019-03-20.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span>
<span class="va">cnv_som_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.cnv.gene.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> |&gt;
  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_cnv_som_gene_process.html">purple_cnv_som_gene_process</a></span><span class="op">(</span>g <span class="op">=</span> <span class="va">umccr_key_genes</span><span class="op">)</span></code></pre></div>
<details><summary>
Description
</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_som_gene</span><span class="op">$</span><span class="va">descr</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per gene) Columns."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Somatic CNVs (per gene) Columns.
</caption>
<thead><tr>
<th style="text-align:left;">
Column
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
gene
</td>
<td style="text-align:left;">
Name of gene
</td>
</tr>
<tr>
<td style="text-align:left;">
minCN/maxCN
</td>
<td style="text-align:left;">
Min/Max copy number found in gene exons
</td>
</tr>
<tr>
<td style="text-align:left;">
chrom/start/end
</td>
<td style="text-align:left;">
Chromosome/start/end location of gene transcript
</td>
</tr>
<tr>
<td style="text-align:left;">
chrBand
</td>
<td style="text-align:left;">
Chromosome band of the gene
</td>
</tr>
<tr>
<td style="text-align:left;">
onco_or_ts
</td>
<td style="text-align:left;">
oncogene (â€˜oncogeneâ€™), tumor suppressor (â€˜tsgeneâ€™), or both (â€˜onco+tsâ€™),
as reported by <a href="https://github.com/jakelever/cancermine" class="external-link">Cancermine</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
transcriptID
</td>
<td style="text-align:left;">
Ensembl transcript ID (dot version)
</td>
</tr>
<tr>
<td style="text-align:left;">
minMinorAlleleCN
</td>
<td style="text-align:left;">
Minimum allele ploidy found over the gene exons - useful for identifying
LOH events
</td>
</tr>
<tr>
<td style="text-align:left;">
somReg (somaticRegions)
</td>
<td style="text-align:left;">
Count of somatic copy number regions this gene spans
</td>
</tr>
<tr>
<td style="text-align:left;">
germDelReg (germlineHomDeletionRegions /
germlineHetToHomDeletionRegions)
</td>
<td style="text-align:left;">
Number of regions spanned by this gene that are (homozygously deleted in
the germline / both heterozygously deleted in the germline and
homozygously deleted in the tumor)
</td>
</tr>
<tr>
<td style="text-align:left;">
minReg (minRegions)
</td>
<td style="text-align:left;">
Number of somatic regions inside the gene that share the min copy number
</td>
</tr>
<tr>
<td style="text-align:left;">
minRegStartEnd
</td>
<td style="text-align:left;">
Start/End base of the copy number region overlapping the gene with the
minimum copy number
</td>
</tr>
<tr>
<td style="text-align:left;">
minRegSupportStartEndMethod
</td>
<td style="text-align:left;">
Start/end support of the CN region overlapping the gene with the min CN
(plus determination method)
</td>
</tr>
</tbody>
</table></details><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_som_gene</span><span class="op">$</span><span class="va">tab</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Somatic CNVs (per gene) Summary Table."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Somatic CNVs (per gene) Summary Table.
</caption>
<thead><tr>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
minCN
</th>
<th style="text-align:right;">
maxCN
</th>
<th style="text-align:left;">
chrom
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:left;">
chrBand
</th>
<th style="text-align:left;">
onco_or_ts
</th>
<th style="text-align:left;">
transcriptID
</th>
<th style="text-align:right;">
minMinorAlleleCN
</th>
<th style="text-align:right;">
somReg
</th>
<th style="text-align:left;">
germDelReg
</th>
<th style="text-align:right;">
minReg
</th>
<th style="text-align:left;">
minRegStartEnd
</th>
<th style="text-align:left;">
minRegSupportStartEndMethod
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
CRBN
</td>
<td style="text-align:right;">
1.0268
</td>
<td style="text-align:right;">
1.0268
</td>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
3150011
</td>
<td style="text-align:right;">
3179710
</td>
<td style="text-align:left;">
p26.2
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000231948.8
</td>
<td style="text-align:right;">
0.0176
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-92214015
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
SDHA
</td>
<td style="text-align:right;">
2.0123
</td>
<td style="text-align:right;">
2.0123
</td>
<td style="text-align:left;">
chr5
</td>
<td style="text-align:right;">
218241
</td>
<td style="text-align:right;">
256700
</td>
<td style="text-align:left;">
p15.33
</td>
<td style="text-align:left;">
tsgene
</td>
<td style="text-align:left;">
ENST00000264932.10
</td>
<td style="text-align:right;">
0.9939
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-48272853
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
DUSP22
</td>
<td style="text-align:right;">
2.0037
</td>
<td style="text-align:right;">
2.0037
</td>
<td style="text-align:left;">
chr6
</td>
<td style="text-align:right;">
292462
</td>
<td style="text-align:right;">
351353
</td>
<td style="text-align:left;">
p25.3
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000419235.6
</td>
<td style="text-align:right;">
0.9868
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-59191910
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
IRF4
</td>
<td style="text-align:right;">
2.0037
</td>
<td style="text-align:right;">
2.0037
</td>
<td style="text-align:left;">
chr6
</td>
<td style="text-align:right;">
391739
</td>
<td style="text-align:right;">
411447
</td>
<td style="text-align:left;">
p25.3
</td>
<td style="text-align:left;">
oncogene
</td>
<td style="text-align:left;">
ENST00000380956.8
</td>
<td style="text-align:right;">
0.9868
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-59191910
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
FOXQ1
</td>
<td style="text-align:right;">
2.0037
</td>
<td style="text-align:right;">
2.0037
</td>
<td style="text-align:left;">
chr6
</td>
<td style="text-align:right;">
1312440
</td>
<td style="text-align:right;">
1314748
</td>
<td style="text-align:left;">
p25.3
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000296839.4
</td>
<td style="text-align:right;">
0.9868
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-59191910
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOCK8
</td>
<td style="text-align:right;">
1.9900
</td>
<td style="text-align:right;">
1.9900
</td>
<td style="text-align:left;">
chr9
</td>
<td style="text-align:right;">
214865
</td>
<td style="text-align:right;">
465259
</td>
<td style="text-align:left;">
p24.3
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000432829.6
</td>
<td style="text-align:right;">
0.9886
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-44377362
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
LARP4B
</td>
<td style="text-align:right;">
1.9899
</td>
<td style="text-align:right;">
1.9899
</td>
<td style="text-align:left;">
chr10
</td>
<td style="text-align:right;">
806914
</td>
<td style="text-align:right;">
931705
</td>
<td style="text-align:left;">
p15.3
</td>
<td style="text-align:left;">
tsgene
</td>
<td style="text-align:left;">
ENST00000612396.4
</td>
<td style="text-align:right;">
0.9825
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-40640101
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
SIRT3
</td>
<td style="text-align:right;">
2.0012
</td>
<td style="text-align:right;">
2.0012
</td>
<td style="text-align:left;">
chr11
</td>
<td style="text-align:right;">
215458
</td>
<td style="text-align:right;">
236431
</td>
<td style="text-align:left;">
p15.5
</td>
<td style="text-align:left;">
tsgene
</td>
<td style="text-align:left;">
ENST00000382743.8
</td>
<td style="text-align:right;">
0.9888
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-52751710
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
KDM5A
</td>
<td style="text-align:right;">
1.9962
</td>
<td style="text-align:right;">
1.9962
</td>
<td style="text-align:left;">
chr12
</td>
<td style="text-align:right;">
280129
</td>
<td style="text-align:right;">
389454
</td>
<td style="text-align:left;">
p13.33
</td>
<td style="text-align:left;">
oncogene
</td>
<td style="text-align:left;">
ENST00000399788.6
</td>
<td style="text-align:right;">
0.9879
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1-35977329
</td>
<td style="text-align:left;">
TELOMERE-CENTROMERE (BAF_WEIGHTED)
</td>
</tr>
<tr>
<td style="text-align:left;">
ZMYM2
</td>
<td style="text-align:right;">
2.0017
</td>
<td style="text-align:right;">
2.0017
</td>
<td style="text-align:left;">
chr13
</td>
<td style="text-align:right;">
19958670
</td>
<td style="text-align:right;">
20091829
</td>
<td style="text-align:left;">
q12.11
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000610343.4
</td>
<td style="text-align:right;">
0.9944
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
0/0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
17025624-31577000
</td>
<td style="text-align:left;">
CENTROMERE-NONE (BAF_WEIGHTED)
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="germline-cnvs-per-chromosome">Germline CNVs (per chromosome)<a class="anchor" aria-label="anchor" href="#germline-cnvs-per-chromosome"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_germ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.cnv.germline.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> |&gt;
  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_cnv_germ_process.html">purple_cnv_germ_process</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<details><summary>
Description
</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_germ</span><span class="op">$</span><span class="va">descr</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Germline CNVs (per chromosome) Columns."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Germline CNVs (per chromosome) Columns.
</caption>
<thead><tr>
<th style="text-align:left;">
Column
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Chr/Start/End
</td>
<td style="text-align:left;">
Coordinates of copy number segment
</td>
</tr>
<tr>
<td style="text-align:left;">
CN
</td>
<td style="text-align:left;">
Fitted absolute copy number of segment adjusted for purity and ploidy
</td>
</tr>
<tr>
<td style="text-align:left;">
CN Min+Maj
</td>
<td style="text-align:left;">
CopyNumber of minor + major allele adjusted for purity
</td>
</tr>
<tr>
<td style="text-align:left;">
Start/End SegSupport
</td>
<td style="text-align:left;">
Type of SV support for the CN breakpoint at start/end of region. Allowed
values: CENTROMERE, TELOMERE, INV, DEL, DUP, BND (translocation), SGL
(single breakend SV support), NONE (no SV support for CN breakpoint),
MULT (multiple SV support at exact breakpoint)
</td>
</tr>
<tr>
<td style="text-align:left;">
Method
</td>
<td style="text-align:left;">
Method used to determine the CN of the region. Allowed values:
BAF_WEIGHTED (avg of all depth windows for the region),
STRUCTURAL_VARIANT (inferred using ploidy of flanking SVs), LONG_ARM
(inferred from the long arm), GERMLINE_AMPLIFICATION (inferred using
special logic to handle regions of germline amplification)
</td>
</tr>
<tr>
<td style="text-align:left;">
BAF (count)
</td>
<td style="text-align:left;">
Tumor BAF after adjusted for purity and ploidy (Count of AMBER baf
points covered by this segment)
</td>
</tr>
<tr>
<td style="text-align:left;">
GC (windowCount)
</td>
<td style="text-align:left;">
Proportion of segment that is G or C (Count of COBALT windows covered by
this segment)
</td>
</tr>
</tbody>
</table></details><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_germ</span><span class="op">$</span><span class="va">tab</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Germline CNVs (per chromosome) Summary Table."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Germline CNVs (per chromosome) Summary Table.
</caption>
<thead><tr>
<th style="text-align:left;">
Chr
</th>
<th style="text-align:right;">
Start
</th>
<th style="text-align:right;">
End
</th>
<th style="text-align:right;">
CN
</th>
<th style="text-align:left;">
CN Min+Maj
</th>
<th style="text-align:left;">
Start/End SegSupport
</th>
<th style="text-align:left;">
Method
</th>
<th style="text-align:left;">
BAF (count)
</th>
<th style="text-align:left;">
GC (windowCount)
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
7510001
</td>
<td style="text-align:right;">
7511000
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:left;">
0+0
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.52 (1)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
14110001
</td>
<td style="text-align:right;">
14113000
</td>
<td style="text-align:right;">
0.4
</td>
<td style="text-align:left;">
0+0.4
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.47 (3)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
15825001
</td>
<td style="text-align:right;">
15829000
</td>
<td style="text-align:right;">
0.4
</td>
<td style="text-align:left;">
0+0.3
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.49 (2)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
58278001
</td>
<td style="text-align:right;">
58279000
</td>
<td style="text-align:right;">
0.3
</td>
<td style="text-align:left;">
0+0.3
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.44 (1)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
61617001
</td>
<td style="text-align:right;">
61618000
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:left;">
0+0.1
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.39 (1)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
79756001
</td>
<td style="text-align:right;">
79757000
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:left;">
0+0
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.35 (1)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
85935001
</td>
<td style="text-align:right;">
85939000
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
0+0.2
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.41 (4)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
89010001
</td>
<td style="text-align:right;">
89013000
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:left;">
0+0.1
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.42 (3)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
105473001
</td>
<td style="text-align:right;">
105481000
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:left;">
0+0.1
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.37 (8)
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:right;">
110834001
</td>
<td style="text-align:right;">
110845000
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:left;">
0+0
</td>
<td style="text-align:left;">
NONE-UNKNOWN
</td>
<td style="text-align:left;">
GERMLINE_HET2HOM_DELETION
</td>
<td style="text-align:left;">
0.98 (0)
</td>
<td style="text-align:left;">
0.33 (9)
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="purity">Purity<a class="anchor" aria-label="anchor" href="#purity"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">purity</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.purity.tsv"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> |&gt;
  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_purity_read.html">purple_purity_read</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">purity</span><span class="op">$</span><span class="va">summary</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Purity Summary Table."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Purity Summary Table.
</caption>
<thead><tr>
<th style="text-align:right;">
n
</th>
<th style="text-align:left;">
variable
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
details
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Purity
</td>
<td style="text-align:left;">
0.75 (0.71-0.78)
</td>
<td style="text-align:left;">
Purity of tumor in the sample (and min-max with score within 10% of
best).
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Ploidy
</td>
<td style="text-align:left;">
1.87 (1.86-1.88)
</td>
<td style="text-align:left;">
Average ploidy of tumor sample after adjusting for purity (and min-max
with score within 10% of best).
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Gender
</td>
<td style="text-align:left;">
MALE
</td>
<td style="text-align:left;">
Gender as inferred by AMBER/COBALT.
</td>
</tr>
<tr>
<td style="text-align:right;">
7
</td>
<td style="text-align:left;">
WGD
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
Whole genome duplication (more than 10 autosomes have average major
allele ploidy &gt; 1.5).
</td>
</tr>
<tr>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
MSI (indels/Mb)
</td>
<td style="text-align:left;">
MSS (0)
</td>
<td style="text-align:left;">
MSI status (MSI, MSS or UNKNOWN if somatic variants not supplied) &amp;
MS Indels per Mb.
</td>
</tr>
<tr>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
PolyclonalProp
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
Proportion of CN regions that are more than 0.25 from a whole CN
</td>
</tr>
<tr>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
DiploidyProp
</td>
<td style="text-align:left;">
0.86 (0.86-0.86)
</td>
<td style="text-align:left;">
Proportion of CN regions that have 1 (+- 0.2) minor and major allele.
</td>
</tr>
<tr>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
TMB
</td>
<td style="text-align:left;">
0 (LOW)
</td>
<td style="text-align:left;">
Tumor mutational burden (# PASS variants per Megabase) (Status: â€˜HIGHâ€™
(&gt;10 PASS per Mb), â€˜LOWâ€™ or â€˜UNKNOWNâ€™).
</td>
</tr>
<tr>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
TML
</td>
<td style="text-align:left;">
0 (LOW)
</td>
<td style="text-align:left;">
Tumor mutational load (# of missense variants) (Status: â€˜HIGHâ€™, â€˜LOWâ€™ or
â€˜UNKNOWNâ€™).
</td>
</tr>
<tr>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
TMB-SV
</td>
<td style="text-align:left;">
713
</td>
<td style="text-align:left;">
# of non inferred, non single passing SVs.
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="kataegis">Kataegis<a class="anchor" aria-label="anchor" href="#kataegis"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.somatic.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="../reference/purple_kataegis.html">purple_kataegis</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">kat</span><span class="op">$</span><span class="va">data</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE Kataegis Table."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE Kataegis Table.
</caption>
<thead><tr>
<th style="text-align:left;">
CHROM
</th>
<th style="text-align:right;">
POS
</th>
<th style="text-align:left;">
KT
</th>
<th style="text-align:left;">
AF
</th>
<th style="text-align:right;">
PURPLE_AF
</th>
<th style="text-align:right;">
PURPLE_CN
</th>
<th style="text-align:right;">
PURPLE_MACN
</th>
<th style="text-align:right;">
PURPLE_VCN
</th>
<th style="text-align:right;">
SUBCL
</th>
<th style="text-align:left;">
MH
</th>
<th style="text-align:left;">
TNC
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
45006212
</td>
<td style="text-align:left;">
REV_1
</td>
<td style="text-align:left;">
0.2143
</td>
<td style="text-align:right;">
0.2601
</td>
<td style="text-align:right;">
1.52
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.396
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
GGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
45007103
</td>
<td style="text-align:left;">
REV_1
</td>
<td style="text-align:left;">
0.1977
</td>
<td style="text-align:right;">
0.2399
</td>
<td style="text-align:right;">
1.52
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.365
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
45007598
</td>
<td style="text-align:left;">
REV_1
</td>
<td style="text-align:left;">
0.24
</td>
<td style="text-align:right;">
0.2913
</td>
<td style="text-align:right;">
1.52
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.444
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr2
</td>
<td style="text-align:right;">
45007709
</td>
<td style="text-align:left;">
REV_1
</td>
<td style="text-align:left;">
0.2273
</td>
<td style="text-align:right;">
0.2759
</td>
<td style="text-align:right;">
1.52
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.420
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71010245
</td>
<td style="text-align:left;">
REV_2
</td>
<td style="text-align:left;">
0.3165
</td>
<td style="text-align:right;">
0.4054
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
0.461
</td>
<td style="text-align:right;">
0.469
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71010553
</td>
<td style="text-align:left;">
REV_2
</td>
<td style="text-align:left;">
0.1863
</td>
<td style="text-align:right;">
0.2387
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
0.461
</td>
<td style="text-align:right;">
0.276
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71011404
</td>
<td style="text-align:left;">
REV_2
</td>
<td style="text-align:left;">
0.2841
</td>
<td style="text-align:right;">
0.3640
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
0.461
</td>
<td style="text-align:right;">
0.421
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
GGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71011410
</td>
<td style="text-align:left;">
REV_2
</td>
<td style="text-align:left;">
0.2727
</td>
<td style="text-align:right;">
0.3494
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
0.461
</td>
<td style="text-align:right;">
0.405
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71011474
</td>
<td style="text-align:left;">
REV_2
</td>
<td style="text-align:left;">
0.2577
</td>
<td style="text-align:right;">
0.3302
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
0.461
</td>
<td style="text-align:right;">
0.382
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71327944
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.1982
</td>
<td style="text-align:right;">
0.2494
</td>
<td style="text-align:right;">
1.26
</td>
<td style="text-align:right;">
0.438
</td>
<td style="text-align:right;">
0.314
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCC
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71327987
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.1947
</td>
<td style="text-align:right;">
0.2450
</td>
<td style="text-align:right;">
1.26
</td>
<td style="text-align:right;">
0.438
</td>
<td style="text-align:right;">
0.309
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71327997
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.2
</td>
<td style="text-align:right;">
0.2517
</td>
<td style="text-align:right;">
1.26
</td>
<td style="text-align:right;">
0.438
</td>
<td style="text-align:right;">
0.317
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71328028
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.15
</td>
<td style="text-align:right;">
0.1888
</td>
<td style="text-align:right;">
1.26
</td>
<td style="text-align:right;">
0.438
</td>
<td style="text-align:right;">
0.238
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71328321
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.1951
</td>
<td style="text-align:right;">
0.2455
</td>
<td style="text-align:right;">
1.26
</td>
<td style="text-align:right;">
0.438
</td>
<td style="text-align:right;">
0.309
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71328545
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.2188
</td>
<td style="text-align:right;">
0.2862
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.302
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCC
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71329130
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.3295
</td>
<td style="text-align:right;">
0.4311
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.455
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
71329696
</td>
<td style="text-align:left;">
FWD_1
</td>
<td style="text-align:left;">
0.3671
</td>
<td style="text-align:right;">
0.4803
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.507
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TCA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78940888
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.2525
</td>
<td style="text-align:right;">
0.3301
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.350
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78941023
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.2589
</td>
<td style="text-align:right;">
0.3385
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.359
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78941779
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.3368
</td>
<td style="text-align:right;">
0.4403
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.467
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
TGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78941859
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.3488
</td>
<td style="text-align:right;">
0.4560
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.483
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
GGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78942061
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.2989
</td>
<td style="text-align:right;">
0.3907
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.414
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
GGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78942101
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.3077
</td>
<td style="text-align:right;">
0.4022
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.426
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78942282
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.3222
</td>
<td style="text-align:right;">
0.4212
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.446
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78942348
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.2892
</td>
<td style="text-align:right;">
0.3780
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.401
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
<tr>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:right;">
78942562
</td>
<td style="text-align:left;">
REV_3
</td>
<td style="text-align:left;">
0.2115
</td>
<td style="text-align:right;">
0.2765
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.392
</td>
<td style="text-align:right;">
0.293
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGA
</td>
</tr>
</tbody>
</table>
<details><summary>
Description
</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">kat</span><span class="op">$</span><span class="va">description</span>, format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"Kataegis column description."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
Kataegis column description.
</caption>
<thead><tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
AF
</td>
<td style="text-align:left;">
Allele Frequency, for each ALT allele, in the same order as listed
</td>
</tr>
<tr>
<td style="text-align:left;">
KT
</td>
<td style="text-align:left;">
Forward/reverse kataegis id
</td>
</tr>
<tr>
<td style="text-align:left;">
MH
</td>
<td style="text-align:left;">
Microhomology
</td>
</tr>
<tr>
<td style="text-align:left;">
PURPLE_AF
</td>
<td style="text-align:left;">
Purity adjusted allelic frequency of variant
</td>
</tr>
<tr>
<td style="text-align:left;">
PURPLE_CN
</td>
<td style="text-align:left;">
Purity adjusted copy number surrounding variant location
</td>
</tr>
<tr>
<td style="text-align:left;">
PURPLE_MACN
</td>
<td style="text-align:left;">
Purity adjusted minor allele ploidy surrounding variant location
</td>
</tr>
<tr>
<td style="text-align:left;">
PURPLE_VCN
</td>
<td style="text-align:left;">
Purity adjusted ploidy of variant
</td>
</tr>
<tr>
<td style="text-align:left;">
SUBCL
</td>
<td style="text-align:left;">
Non-zero subclonal likelihood
</td>
</tr>
<tr>
<td style="text-align:left;">
TNC
</td>
<td style="text-align:left;">
Tri-nucleotide context
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="qc">QC<a class="anchor" aria-label="anchor" href="#qc"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/purple/purple.qc"</span>, package <span class="op">=</span> <span class="st">"gpgr"</span><span class="op">)</span> |&gt;
  <span class="fu">gpgr</span><span class="fu">::</span><span class="fu"><a href="../reference/purple_qc_read.html">purple_qc_read</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">qc</span><span class="op">$</span><span class="va">summary</span> |&gt;
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>format <span class="op">=</span> <span class="st">"html"</span>, caption <span class="op">=</span> <span class="st">"PURPLE QC Summary Table."</span><span class="op">)</span></code></pre></div>
<table class="table">
<caption>
PURPLE QC Summary Table.
</caption>
<thead><tr>
<th style="text-align:right;">
n
</th>
<th style="text-align:left;">
variable
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
details
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
QC_Status
</td>
<td style="text-align:left;">
WARN_DELETED_GENES
</td>
<td style="text-align:left;">
See â€˜Descriptionâ€™.
</td>
</tr>
<tr>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
Method
</td>
<td style="text-align:left;">
NORMAL
</td>
<td style="text-align:left;">
Fit method (NORMAL, HIGHLY_DIPLOID, SOMATIC or NO_TUMOR).
</td>
</tr>
<tr>
<td style="text-align:right;">
14
</td>
<td style="text-align:left;">
CopyNumberSegments
</td>
<td style="text-align:left;">
1428 (Unsupported: 0)
</td>
<td style="text-align:left;">
# of CN segments.
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Purity
</td>
<td style="text-align:left;">
0.8600
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:right;">
17
</td>
<td style="text-align:left;">
Gender
</td>
<td style="text-align:left;">
Amber: MALE; Cobalt: MALE
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:right;">
14
</td>
<td style="text-align:left;">
DeletedGenes
</td>
<td style="text-align:left;">
7782
</td>
<td style="text-align:left;">
# of homozygously deleted genes.
</td>
</tr>
<tr>
<td style="text-align:right;">
15
</td>
<td style="text-align:left;">
Contamination
</td>
<td style="text-align:left;">
0.0
</td>
<td style="text-align:left;">
Rate of contamination in tumor sample as determined by AMBER.
</td>
</tr>
<tr>
<td style="text-align:right;">
16
</td>
<td style="text-align:left;">
GermlineAberrations
</td>
<td style="text-align:left;">
NONE
</td>
<td style="text-align:left;">
Can be one or more of: KLINEFELTER, TRISOMY_X/21/13/18/15, XYY,
MOSAIC_X.
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<table class="table">
<caption>Main packages used in this vignette.</caption>
<thead><tr class="header">
<th align="left">package</th>
<th align="left">version</th>
<th align="left">datestamp</th>
<th align="left">source</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">base</td>
<td align="left">4.1.2</td>
<td align="left">2022-01-16</td>
<td align="left">local</td>
</tr>
<tr class="even">
<td align="left">gpgr</td>
<td align="left">1.2.1</td>
<td align="left">2022-02-09</td>
<td align="left">local</td>
</tr>
</tbody>
</table>
<table class="table">
<caption>Platform information.</caption>
<colgroup>
<col width="12%">
<col width="88%">
</colgroup>
<thead><tr class="header">
<th align="left">name</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">version</td>
<td align="left">R version 4.1.2 (2021-11-01)</td>
</tr>
<tr class="even">
<td align="left">os</td>
<td align="left">Ubuntu 20.04.3 LTS</td>
</tr>
<tr class="odd">
<td align="left">system</td>
<td align="left">x86_64, linux-gnu</td>
</tr>
<tr class="even">
<td align="left">ui</td>
<td align="left">X11</td>
</tr>
<tr class="odd">
<td align="left">language</td>
<td align="left">en</td>
</tr>
<tr class="even">
<td align="left">collate</td>
<td align="left">C.UTF-8</td>
</tr>
<tr class="odd">
<td align="left">ctype</td>
<td align="left">C.UTF-8</td>
</tr>
<tr class="even">
<td align="left">tz</td>
<td align="left">Etc/UTC</td>
</tr>
<tr class="odd">
<td align="left">date</td>
<td align="left">2022-02-09</td>
</tr>
<tr class="even">
<td align="left">pandoc</td>
<td align="left">2.17.1.1 @ /usr/share/miniconda/envs/pkgdown/bin/ (via
rmarkdown)</td>
</tr>
</tbody>
</table>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://github.com/pdiakumis" class="external-link">Peter
Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
